src/Compilers/CSharp/Test/Emit3/Semantics/CollectionExpressionTests_WithElement_Extra.cs

```diff
@@ -1520,6 +1645,51 @@ static void Main()
             Diagnostic(ErrorCode.ERR_CollectionBuilderAttributeMethodNotFound, "params MyCollection<T> c").WithArguments("Create", "T", "MyCollection<T>").WithLocation(12, 38));
     }
 
+    [Fact]
+    public void CollectionBuilder_OptionalParameter_A()
+    {
+        string sourceA = """
+                using System;
+                using System.Collections;
+                using System.Collections.Generic;
+                using System.Runtime.CompilerServices;
+                [CollectionBuilder(typeof(MyBuilder), "Create")]
+                class MyCollection<T> : IEnumerable<T>
+                {
+                    public readonly T Arg;
+                    public MyCollection(T arg, ReadOnlySpan<T> items) { Arg = arg; }
+                    IEnumerator<T> IEnumerable<T>.GetEnumerator() => null;
+                    IEnumerator IEnumerable.GetEnumerator() => null;
+                }
+                class MyBuilder
+                {
+                    public static MyCollection<T> Create<T>(T arg = default, ReadOnlySpan<T> items = default) => new(arg, items);
+                }
+                """;
+        string sourceB = """
+                using System;
+                class Program
+                {
+                    static void Main()
+                    {
+                        Console.WriteLine(EmptyArgs<int>().Arg);
+                        Console.WriteLine(NonEmptyArgs(2).Arg);
+                        Console.WriteLine(Params(3, 4).Arg);
+                    }
+                    static MyCollection<T> EmptyArgs<T>() => [with()];
+                    static MyCollection<T> NonEmptyArgs<T>(T t) => [with(t)];
+                    static MyCollection<T> Params<T>(params MyCollection<T> c) => c;
```

Comment:
Consider splitting this test into the portion that works and can run, and the one case that errors